File: src/main/java/com/ject/studytrip/studylog/presentation/dto/response/CreateStudyLogResponse.java
package com.ject.studytrip.studylog.presentation.dto.response;

import com.ject.studytrip.studylog.application.dto.StudyLogInfo;
import io.swagger.v3.oas.annotations.media.Schema;

public record CreateStudyLogResponse(@Schema(description = "생성된 학습 로그 ID") Long studyLogId) {
    public static CreateStudyLogResponse of(StudyLogInfo info) {
        return new CreateStudyLogResponse(info.studyLogId());
    }
}

File: src/main/kotlin/com/ject/studytrip/studylog/presentation/dto/response/CreateStudyLogResponse.kt
package com.ject.studytrip.studylog.presentation.dto.response

import com.ject.studytrip.studylog.application.dto.StudyLogInfo
import io.swagger.v3.oas.annotations.media.Schema

data class CreateStudyLogResponse(
    @field:Schema(description = "생성된 학습 로그 ID")
    val studyLogId: Long,
) {
    companion object {
        @JvmStatic
        fun of(studyLogInfo: StudyLogInfo): CreateStudyLogResponse = CreateStudyLogResponse(studyLogInfo.studyLogId)
    }
}
